ledger/complete/mtrie/forest_test.go

```diff
@@ -889,14 +912,32 @@ func TestProofGenerationInclusion(t *testing.T) {
 	paths := []ledger.Path{p1, p2, p3, p4}
 	payloads := []*ledger.Payload{v1, v2, v3, v4}
 
+	// shuffle the order of `proofPaths` to run `Proofs` on non-sorted input paths
+	rand.Shuffle(len(paths), func(i, j int) {
+		paths[i], paths[j] = paths[j], paths[i]
+	})
```

Comment:
what is the reason behind this shuffle?, we need to always sort the proofs by paths as we need the chunk data pack ID to be deterministic.

Reply:
With this PR, the `Proofs` function sorts the input paths. In the test, I want to make sure the sorting has happened, whatever the initial order of the input paths is. That's why the test shuffles the paths before calling `Proofs`.

Reply:
Cơ sở tự động thay đổi từ tarak / 5346 -mize-mtrie-algos đến bậc thầy Hôm kia

Reply:
I see